harness/determined/pytorch/dsat/_dsat_search_method.py

```diff
@@ -907,6 +924,9 @@ def get_trial_list_after_successful_run(
         return [trial]
 
     def should_stop_lineage(self, trial: DSATTrial) -> bool:
+        # TODO: This breaks the tests but perhaps it's how we should be doing this?
```

Comment:
you mean the e2e tests? which ones?

Also, did you mean to comment out the assert below?

Reply:
I did, it messes with the unit tests, and I don't fully know why or how and figured it would be better to leave that for you to try

Reply:
oh, you meant the commented out `assert` breaks the unit tests. Thought you meant the whole `should_stop_lineage` function was somehow breaking e2e tests or something, got it.